### DISCUSSION TITLE
URL with Git branch :  Does this actually work

### DISCUSSION THREAD

**Discussion by willTastyMakers** [vercel/community#2305] [Help]
### Summary

I found someone asked a similar question [How to get the autogenerated git branch url from vercel?](https://github.com/orgs/vercel/discussions/1652) However, I was unable to discern the answer given by @[amyegan](https://github.com/amyegan). 

The [URL with Git branch](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) link described by @[amyegan](https://github.com/amyegan) as the  _always up-to-date branch_ url either does not work else I am just not perceiving the description correctly.

1.  I have a  Preview deploy to  branch called "dev" :  [  tms-nebula-cyfit76rz-wstreetertms.vercel.app](https://tms-nebula-cyfit76rz-wstreetertms.vercel.app/)

![Screenshot 2023-05-04 at 1 06 40 PM](https://user-images.githubusercontent.com/58872627/236318031-fc69dabb-08f7-4b66-8053-4150eee5b81b.png)

2. [URL with Git branch](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) describes the url structure for  _always up-to-date branch_ url   as 
      `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`
      
      
3. My assessment is 
     project-name:  "tms-nebula" 
     branch-name: "dev"
     scope-slug: "wstreetertms"
     
     which suggest to me the url _always up-to-date branch_ url  =    https://tms-nebula-git-dev-wstreetertms.vercel.app   
     
     However, this has yet to work for any of my preview branches ( stage, dev)
     
     
If you could shed some light and what I am doing wrong or how I have misinterpreted these URL with Git Branch Instructions, it would be very helpful.

Thanks you

**Comment by amyegan** [comment]
Sorry you're having trouble with this, @willTastyMakers. The always-up-to-date git branch URL should be added automatically when you deploy. I need to get some more information to help with debugging. Please answer as many of the following as you can.
- Are you using the automated [Vercel GitHub integration](https://vercel.com/docs/concepts/deployments/git/vercel-for-github), the [CLI](https://vercel.com/docs/cli), or another method to deploy the site?
- Are you using a vercel.json config? If so, what properties are used?
- What framework (e.g. Next.js, Nuxt, Svelte) is used for your site?
- Do you have any other code, settings, or configuration related to domains?

**Reply by willTastyMakers** [reply]
Hi @amyegan ,

Thank you for the speedy reply. 


###  **Are you using the automated [Vercel GitHub integration](https://vercel.com/docs/concepts/deployments/git/vercel-for-github), the [CLI](https://vercel.com/docs/cli), or another method to deploy the site?**
 
 When everything is set up it would be great to control the deploy with the vercel CLI through github workflow actions, that being said. 
 

1.   My initial setup was only using the github Vercel App, and simply allowing vercel to deploy on any pushes to main, stage, and dev branches. It deployed to vercel correctly and the preview sites appear  correctly

 
 `tms-nebula-<hash #>-wstreetertms.vercel.app` |  ex,  https://tms-nebula-cyfit76rz-wstreetertms.vercel.app
 
   but  
   
`tms-nebula-git-<dev or stage>-wstreetertms.vercel.app`  |  ex,  https://tms-nebula-git-dev-wstreetertms.vercel.app
    
    
 did not work and simply retuned an error page. 

 
2.  The next setup was to remove the vercel app from github completely and create a project through vercel dashboard and using the `vercel cli`, to create  a  _project-id_ and an  _org-id_ ,  integrating them as github secrets used in the github workflow actions  to upload the code to  vercel through github worklfow actions. This set up did not work as a connection could not be made to vercel.
 
3.  For my  third  attempt,  I re-installed the vercel app in github  and created a new project in the vercel Dashboard associating it with my github repository, and I created a new  _project-id_ and an   _org-id_  with the vercel cli.  My git hub worklow actions were successful, however. same result as my  initial set up;
 -   `tms-nebula-<hash #>-wstreetertms.vercel.app` WORKS
 -   `tms-nebula-git-<dev or stage>-wstreetertms.vercel.app`   DOES NOT WORK

 
### **Are you using a vercel.json config? If so, what properties are used?**
 
 Yes I am using a vercel.json config
  
```
  {
  "github": {
    "main": false,
    "stage": false,
    "dev": false
  }
}
```

### **What framework (e.g. Next.js, Nuxt, Svelte) is used for your site?**
 
 Next.js ( 13.4.1).  ( Basically using  [vercel/commerce ex.](https://github.com/vercel/commerce) )


### **Do you have any other code, settings, or configuration related to domains?**

The only other api call to a domain I have is to Shopify. I also have the Shopify app installed in vercel account


Thank you ,  
@amyegan  
( Go Dubs... )

**Reply by willTastyMakers** [reply]
Hi @amyegan,
I can see  you are crazy busy, but any help on this issue would be much appreciated.

Thanks  🙏 in advance.

**Reply by amyegan** [reply]
Sorry for the delay. And thanks for understanding and for your patience. I still haven't worked out why the automatic alias isn't working, but you have another option since you're already using the CLI to deploy with GitHub Actions. 

You can add your own alias with [`vercel alias`](https://vercel.com/docs/cli/alias#) on your preview deployments.

**Comment by kyle-turn2** [comment]
@amyegan My Branch URLs are not working either (using the automated GitHub integration for preview and production branches). I just get a "This site can't provide a secure connection" with the ERR_SSL_PROTOCOL_ERROR message. Are the certificates getting issued correctly?

**Reply by kyle-turn2** [reply]
Nevermind, something happened to that particular branch because when I created a new one all is good.

**Comment by willTastyMakers** [comment]
Hi Amy,
When I set up the deployment for Nextjs Website under an paid organization account, I have the ability to deploy automatically to "dev", "stage" and production.
